src/main/groovy/com/github/benmanes/gradle/versions/updates/Resolver.groovy

```diff
@@ -416,7 +418,7 @@ class Resolver {
   }
 
   private boolean supportsConstraints(Configuration configuration) {
```

Comment:
Can you add a little JavaDoc explaining the `fromBuildSrc` condition so that the rational is easier to recall for future readers?

Reply:
I will definitely add some javadoc and comment ^^
Do you think fromBuildSrc should be a parameter like checkConstraints ?

Reply:
Or rather, should there be a parameter like ignoreConstraintsFromBuildSrc at the same level as checkConstraints ?

Reply:
If we did that, I’d do it as an enablement flag. Then you just pass in a check constraints boolean for each, e.g. checkBuildEnvironmentConstraints, and you don’t need a suppression flag. How about that?

Reply:
Hum, I'm not sure what you meant. Something like this ?
```
tasks.withType<DependencyUpdatesTask> {
  checkProjectsConstraints = true
  checkBuildEnvironmentConstraints = true
}
```

Reply:
Yes. You don't need to pass `fromBuildSrc` into `Resolver` as it only needs `checkConstraints`, which can be determined at the task level. That configuration example is simpler to reason about and then is passing the param value through from the task to the resolver. For backwards compatibility I'd keep `checkProjectsConstraints` called `checkConstraints` or keep it as an alias it if you prefer the clearer terminology.

Reply:
(Also this is optional based on if you prefer it or now. I'm fine merging as is or having this new configuration option)

Reply:
I would rather let the user have the option to check those constraints, not all of them will come from gradle.

Reply:
I added the configuration option, it feel way better than just ignoring those external constraints :)
I had to adapt the tests, since a new log4j vulnerability was patched.
There is something bothering about the test, I think I should not have seen an update to log4j-core:2.17.1 since i specified only 2.16.0 and 2.17.0 in the test arborescence, so I must be missing something

Reply:
FWIU the tests are intended to run against the mock m2 repository but I wonder if the mock is only applied to the project dependencies and not the build ones? I guess so since it's a hard problem - Gradle can easily add more build dependencies in tbe future and maintaining a mock m2 for Gradle seems challenging. I personally think the current assertions, checking that an update was presented, but not the actual presented version, make sense.